### PR TITLE
rm 'dangerous_fn_linter' to fix #173

### LINKED ISSUE #173: dangerous_functions_linter could be replaced by lintr::undesirable_function_linter()
https://github.com/ropensci-review-tools/goodpractice/blob/b215780eac3f39ff318ebd6d835ff58b6cd18880/R/my_linters.R#L111

https://lintr.r-lib.org/reference/undesirable_function_linter.html

Again similar to #171/#172, this is `undesirable_function_linter(funcs, symbol_is_undesirable=FALSE)`.

### PR DESCRIPTION
**Pull request by mpadge**

### CODE REVIEW

File: R/prep_lintr.R [hunk 1]
```diff
@@ -5,10 +5,20 @@ linters_to_lint <- list(
...
   trailing_semicolon_linter = trailing_semicolon_linter(),
-  attach_detach_linter = attach_detach_linter(),
-  setwd_linter = setwd_linter(),
-  sapply_linter = sapply_linter(),
-  library_require_linter = library_require_linter(),
+  attach_detach_linter = lintr::undesirable_function_linter(fun = c(
+    "attach" = "Avoid attach, it is easy to create errors with it",
+    "detach" = "Avoid detach, it is easy to create errors with it"
+  )),
+  setwd_linter = lintr::undesirable_function_linter(fun = c(
```
Comment (maelle) [inline R/prep_lintr.R:12]: I think you might be able to rely on the defaults https://lintr.r-lib.org/reference/default_undesirable_functions.html

**Review by maelle — CHANGES_REQUESTED** [review]
So nice to be able to use more of lintr. 

In the PR first comment why not use the "Fix #" syntax to link the PR to the issue? https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests#linking-a-pull-request-to-an-issue

**Comment by mpadge** [comment]
> So nice to be able to use more of lintr.
> 
> In the PR first comment why not use the "Fix #" syntax to link the PR to the issue? [docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests#linking-a-pull-request-to-an-issue](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests#linking-a-pull-request-to-an-issue)

Sorry, I almost always do that, but was feeling pretty groggy yesterday and wasn't thinking quite right ... But thanks for always catching me out :laughing: :exclamation:

File: R/prep_lintr.R (hunk 1 above)
Comment (mpadge) [reply R/prep_lintr.R:12]: Yeah, I pondered that, and it is of course possible, but the current setup has this useful explicit list here:
https://github.com/ropensci-review-tools/goodpractice/blob/c3b60fb3f6be4ffae2a835631cb27119f166db12/R/prep_lintr.R#L4-L13
Bundling all in single default `undesirable_functions` would then require people to look somewhere in `lintr` to find out which default functions were considered undesirable. I like this explicit naming more. Does that make sense?

Comment (maelle) [reply R/prep_lintr.R:12]: But would there be a way for us not to store the messages here? Like for instance having `"Avoid changing the working directory, or restore it in on.exit"` in goodpractice source.

Comment (mpadge) [reply R/prep_lintr.R]: Yes, of course there is! Thank you so much for pushing further. Code in current form is much neater and more robust. thanks!

**Review by maelle — APPROVED** [review]

**Pull request merged**